lib/ngx/ssl.lua

```diff
@@ -169,6 +184,80 @@ local charpp = ffi.new("char*[1]")
 local intp = ffi.new("int[1]")
 
 
+function _M.client_server_name()
+    if subsystem ~= 'http' then
+        error("no support stream")
+    end
+
+    local r = get_request()
+    if not r then
+        error("no request found")
+    end
+
+    local sizep = get_size_ptr()
+
+    local rc = ngx_lua_ffi_ssl_client_server_name(r, charpp, sizep, errmsg)
+    if rc == FFI_OK then
+        return ffi_str(charpp[0], sizep[0])
+    end
+
+    return nil, ffi_str(errmsg[0])
+end
+
+do
+    local protocal_flags = {
+        ["SSLv2"]   = 0x0002,
```

Comment:
since we have change the c code, could you set these args like `ngx.ERR`, user can use `set_protocols` like `set_protocols(ngx.SSL_TLSv11)`